File: samples/snippets/csharp/System.IO.Compression/Deflate/FileCompressCompressionLevelExample.cs
//<snippet1>
using System;
using System.IO;
using System.IO.Compression;

public static class FileCompressCompressionLevelExample
{
    private const string OriginalFileName = "original.txt";
    private const string CompressedFileName = "compressed.dfl";

    public static void Main()
    {
        CreateFileToCompress();
        CompressFile();
        PrintResults();
        DeleteFiles();

        /*
         Output:
            The original file 'original.txt' weighs 55 bytes. Contents: "This is the content saved inside the original.txt file."
            The compressed file 'compressed.dfl' weighs 54 bytes.
         */
    }

    private static void CreateFileToCompress()
    {
        using var writer = new StreamWriter(OriginalFileName);
        writer.Write($"This is the content saved inside the {OriginalFileName} file.");
    }

    private static void CompressFile()
    {
        using FileStream originalFileStream = File.Open(OriginalFileName, FileMode.Open);
        using FileStream compressedFileStream = File.Create(CompressedFileName);
        using var deflateStream = new DeflateStream(compressedFileStream, CompressionLevel.Fastest);
        originalFileStream.CopyTo(deflateStream);
    }

    private static void PrintResults()
    {
        long originalSize = new FileInfo(OriginalFileName).Length;
        long compressedSize = new FileInfo(CompressedFileName).Length;
        Console.WriteLine($"The original file '{OriginalFileName}' weighs {originalSize} bytes. Contents: \"{File.ReadAllText(OriginalFileName)}\"");
        Console.WriteLine($"The compressed file '{CompressedFileName}' weighs {compressedSize} bytes.");
    }

    private static void DeleteFiles()
    {
        File.Delete(OriginalFileName);
        File.Delete(CompressedFileName);
    }
}
//</snippet1>

File: samples/snippets/visualbasic/System.IO.Compression/Deflate/FileCompressCompressionLevelExample.vb
'<snippet1>
Imports System.IO
Imports System.IO.Compression

Module FileCompressCompressionLevelExample
    Private Const OriginalFileName As String = "original.txt"
    Private Const CompressedFileName As String = "compressed.dfl"

    Sub Main()
        CreateFileToCompress()
        CompressFile()
        PrintResults()
        DeleteFiles()

        'Output:
        '   The original file 'original.txt' weighs 55 bytes. Contents: "This is the content saved inside the original.txt file."
        '   The compressed file 'compressed.dfl' weighs 54 bytes.
    End Sub

    Private Sub CreateFileToCompress()
        Using writer = New StreamWriter(OriginalFileName)
            writer.Write($"This is the content saved inside the {OriginalFileName} file.")
        End Using
    End Sub

    Private Sub CompressFile()
        Using originalFileStream As FileStream = File.Open(OriginalFileName, FileMode.Open)
            Using compressedFileStream As FileStream = File.Create(CompressedFileName)
                Using deflateStream = New DeflateStream(compressedFileStream, CompressionLevel.Fastest)
                    originalFileStream.CopyTo(deflateStream)
                End Using
            End Using
        End Using
    End Sub

    Private Sub PrintResults()
        Dim originalSize As Long = New FileInfo(OriginalFileName).Length
        Dim compressedSize As Long = New FileInfo(CompressedFileName).Length
        Console.WriteLine($"The original file '{OriginalFileName}' weighs {originalSize} bytes. Contents: ""{File.ReadAllText(OriginalFileName)}""")
        Console.WriteLine($"The compressed file '{CompressedFileName}' weighs {compressedSize} bytes.")
    End Sub

    Private Sub DeleteFiles()
        File.Delete(OriginalFileName)
        File.Delete(CompressedFileName)
    End Sub
End Module
'</snippet1>
